tests/validator/validator.cpp

```diff
@@ -608,20 +605,67 @@ TEST(Validator, invalidMathMLVariables)
     EXPECT_EQ_ERRORS(expectedErrors, v);
 }
 
+TEST(Validator, invalidSimpleMathmlCellMLUnits)
+{
+    const std::string math =
+        "<math  xmlns:cellml=\"http://www.cellml.org/cellml/2.0#\" xmlns=\"http://www.w3.org/1998/Math/MathML\"><apply><bvar><ci cellml:units=\"dimensionless\">B</ci></bvar></apply></math>";
+    const std::vector<std::string> expectedErrors = {
+        "CellML identifiers must contain one or more basic Latin alphabetic characters.",
+        "Model does not have a valid name attribute.",
+        "CellML identifiers must contain one or more basic Latin alphabetic characters.",
+        "Component does not have a valid name attribute.",
```

Comment:
Would it be practical to start reporting components with relationship to their parents now that we can do that easily?  It would make it a lot easier to locate the source of the error ... eg: "Component inside parent component "somethingHere" does not have a valid name" etc?

Reply:
No, the error itself holds a reference to the component that has the error.

Reply:
.. but the user doesn't get to see it unless they know to go hunting?  If you're interacting with libCellML only though a basic C++ file then this kind of error does nothing to help you.